File: sample/ruby/main_animate_global_sample.rb
require 'voxelamming'
# require_relative 'voxelamming'

room_name = "1000"
vox = Voxelamming::VoxelammingManager.new(room_name)
# vox = VoxelammingManager.new(room_name)

vox.set_box_size(0.3)
vox.set_build_interval(0.01)

(0..9).each do |i|
  vox.create_box(-1, i, 0, r: 0, g: 1, b: 1, alpha: 1)
  vox.create_box(0, i, 0, r: 1, g: 0, b: 0, alpha: 1)
  vox.create_box(1, i, 0, r: 1, g: 1, b: 0, alpha: 1)
  vox.create_box(2, i, 0, r: 0, g: 1, b: 1, alpha: 1)
end

(0..4).each do |i|
  vox.remove_box(0, i * 2 + 1, 0)
  vox.remove_box(1, i * 2, 0)
end

node_positions = [
  [0, 0, 0],
  [-10, 0, 0],
  [10, 0, 0],
  [0, -20, 0],
  [0, 20, 0],
  [0, 0, -10],
  [0, 0, 10]
]

node_positions.each do |x, y, z|
  vox.transform(x, y, z, pitch: 0, yaw: 0, roll: 0)
  vox.send_data()
  sleep(0.1)
end

vox.animate_global(0, 0, 0, pitch: 0, yaw: 180, roll: 0, scale: 1, interval: 100)
vox.send_data()

File: sample/python/main_animate_global_sample.py
from time import sleep
# voxelammingパッケージからVoxelammingクラスをインポートします
from voxelamming import Voxelamming
# from voxelamming_local import Voxelamming  # ローカルで開発している場合はこちらを使う

# Voxelammingアプリに表示されている部屋名を指定してください
room_name = "1000"
# Voxelammingクラスのインスタンスを生成します
voxelamming = Voxelamming(room_name)
# ボクセルの設定を行います
voxelamming.set_box_size(0.3)
voxelamming.set_build_interval(0.01)

# ボクセルを配置するため、位置と色を設定します
for i in range(10):
    voxelamming.create_box(-1, i, 0, r=0, g=1, b=1, alpha=1)
    voxelamming.create_box(0, i, 0, r=1, g=0, b=0, alpha=1)
    voxelamming.create_box(1, i, 0, r=1, g=1, b=0, alpha=1)
    voxelamming.create_box(2, i, 0, r=0, g=1, b=1, alpha=1)

for i in range(5):
    voxelamming.remove_box(0, i * 2 + 1, 0)
    voxelamming.remove_box(1, i * 2, 0)

# ボクセルを配置する位置を設定します
node_positions = [
    [0, 0, 0],
    [-10, 0, 0],
    [10, 0, 0],
    [0, -20, 0],
    [0, 20, 0],
    [0, 0, -10],
    [0, 0, 10]

]

for x, y, z in node_positions:
    # ボクセルを配置するため、位置を設定します
    voxelamming.transform(x, y, z, pitch=0, yaw=0, roll=0)
    # ボクセルデータをアプリに送信します。（位置を変えて、複数回送信）
    voxelamming.send_data()
    # 1秒待機します
    sleep(0.1)

voxelamming.animate_global(0, 0, 0, pitch=0, yaw=180, roll=0, scale=1, interval=100)

# ボクセルデータをアプリに送信します。（グローバルアニメーション）
voxelamming.send_data()
